src/main/java/mate/academy/spring/dao/impl/RoleDaoImpl.java

```diff
@@ -0,0 +1,31 @@
+package mate.academy.spring.dao.impl;
+
+import java.util.Optional;
+import mate.academy.spring.dao.AbstractDao;
+import mate.academy.spring.dao.RoleDao;
+import mate.academy.spring.exception.DataProcessingException;
+import mate.academy.spring.model.Role;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class RoleDaoImpl extends AbstractDao<Role> implements RoleDao {
+    public RoleDaoImpl(SessionFactory factory) {
+        super(factory, Role.class);
+    }
+
+    @Override
+    public Optional<Role> getByName(String roleName) {
+        try (Session session = factory.openSession()) {
+            Role.RoleName roleName1 = Role.RoleName.valueOf(roleName);
+            Query<Role> getByName = session.createQuery(
+                    "FROM Role WHERE roleName = :roleName", Role.class);
+            getByName.setParameter("roleName", roleName1);
+            return getByName.uniqueResultOptional();
+        } catch (Exception e) {
+            throw new DataProcessingException("Role with name" + roleName + " not found", e);
```

Comment:
The exception message is missing a space between 'name' and the roleName, which will cause the error message to look odd. It's a small issue but good to fix for better readability of the error messages.